### PR TITLE
Using VersionedClient for SeekableByteChannel Tests

### PR DESCRIPTION
**Pull request by ibrahimrabab**
# Description

Using existing versioned service client to allow live tests to pass.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobSeekableByteChannelTests.groovy [hunk 1]
```diff
@@ -21,10 +22,17 @@ import java.nio.ByteBuffer
 import java.time.Duration
 
 class BlobSeekableByteChannelTests extends APISpec {
+    BlobContainerClient versionedCC
     BlobClient bc
 
     def setup() {
-        bc = cc.getBlobClient(generateBlobName())
+        versionedCC = versionedBlobServiceClient.getBlobContainerClient(generateContainerName())
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobSeekableByteChannelTests.groovy:29]: All tests in this test class will need to be regenerated as there is an additional recorded random string being generated due to the `generateContainerName()` call.

**Review by jaschrep-msft — CHANGES_REQUESTED** [review]
There's only one test that needs versioned clients. We don't need to make every test here require a versioning account. It would be better to just have this setup in the test that needs it.

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**